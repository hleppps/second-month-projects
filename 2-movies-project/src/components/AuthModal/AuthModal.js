import { useRef, useState } from "react";
import { Button, Form, Modal, Spinner } from "react-bootstrap";
import styles from "./AuthModal.module.scss";

const AuthModal = (props) => {
  const [isFetching, setIsFetching] = useState(false);
  const [isError, setIsError] = useState(false);
  const loginInputRef = useRef();
  const passwordInputRef = useRef();

  const onSubmitHadler = async (event) => {
    event.preventDefault();
    setIsError(false);
    setIsFetching(true);

    await props.authenticationHandler(
      loginInputRef,
      passwordInputRef,
      setIsError,
      setIsFetching
    );
  };

  return (
    <>
      <Modal show={props.modalIsShown} onHide={props.onHideModal}>
        <Modal.Header closeButton>
          <Modal.Title>Authorization</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form data-testid="login-form" onSubmit={onSubmitHadler}>
            <Form.Group className="mb-3" controlId="formUsername">
              <Form.Label>Username</Form.Label>
              <Form.Control
                type="text"
                defaultValue="hlep"
                required
                ref={loginInputRef}
              />
            </Form.Group>
            <Form.Group className="mb-3" controlId="formPassword">
              <Form.Label>Password</Form.Label>
              <Form.Control
                type="password"
                defaultValue="121212"
                required
                ref={passwordInputRef}
              />
            </Form.Group>
            <Form.Group className={styles.footer}>
              {!isFetching && (
                <Button className="w-100" type="submit" variant="outline-dark">
                  Sign in
                </Button>
              )}
              {isFetching && <Spinner className="mx-10" animation="grow" />}
              {!isFetching && isError && (
                <p>Invalid username and/or password</p>
              )}
            </Form.Group>
          </Form>
        </Modal.Body>
      </Modal>
    </>
  );
};

export default AuthModal;
