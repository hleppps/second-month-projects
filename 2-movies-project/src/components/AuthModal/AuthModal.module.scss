.footer {
  display: flex;
  flex-direction: column;
}

.footer p {
  margin: 0.4rem 0 0 0;
  color: rgb(255, 0, 0);
}
