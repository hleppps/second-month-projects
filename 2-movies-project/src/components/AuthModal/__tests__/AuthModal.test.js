import { cleanup, fireEvent, render } from "@testing-library/react";
import AuthModal from "../AuthModal";

afterEach(cleanup);

describe("AuthModal component", () => {
  it("calls the authHandler function", () => {
    const authHandler = jest.fn();
    const modalIsShown = true;

    const { getByTestId } = render(
      <AuthModal
        authenticationHandler={authHandler}
        modalIsShown={modalIsShown}
      />,
      {}
    );

    fireEvent.submit(getByTestId("login-form"));
    expect(authHandler).toBeCalled();
  });
  
  it("matches a snapshot", () => {
    const { asFragment } = render(<AuthModal />);
    expect(asFragment(<AuthModal />)).toMatchSnapshot();
  });
});
