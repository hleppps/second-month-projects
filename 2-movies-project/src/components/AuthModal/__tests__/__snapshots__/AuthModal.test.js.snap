// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModal component matches a snapshot 1`] = `<DocumentFragment />`;
